modules/calc/src/main/java/com/opengamma/strata/calc/runner/Results.java

```diff
@@ -119,6 +131,23 @@ private String invalidColumnIndexMessage(int columnIndex) {
         columnIndex);
   }
 
+  //-------------------------------------------------------------------------
+  /**
+   * Processes a successful result by applying a function that alters each value.
+   * <p>
+   * This operation allows post-processing of result values.
+   * The specified function represents a conversion to be performed on each value.
+   *
+   * @param function  the function to transform the value with
+   * @return the new results
+   */
+  public Results map(Function<Result<?>, Result<?>> function) {
```

Comment:
public Results calculateSingleScenario(
        CalculationTasks tasks,
        CalculationEnvironment marketData,
        ReferenceData refData) {

      // perform the calculations
      Results results = calculateMultipleScenarios(tasks, marketData, refData);

      // unwrap the results
      // since there is only one scenario it is not desirable to return scenario result containers
      return results.map(r -> unwrapScenarioResult(r));
    }